### ISSUE TITLE
Roslyn incorrectly reports empty property initializers as possible dereferences

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#38060]
**Version Used**: 

```
Branch master (13 Aug 2019)
Latest commit a08336 by David:
Merge pull request #37916 from dibarbet/partial_load_str

Update partial load dialog string to better match task center.
```

**Steps to Reproduce**:

Compile adn run the following code
```cs
#nullable enable
class C
{
  C? f;
    
  static void Main()
  {
    new C() { f = {  }};
  }
}
```

**Expected Behavior**:
No warnings. The field `f` will not be dereferened since there's nothing to initialize here.

**Actual Behavior**:
`warning CS8602: Dereference of a possibly null reference.` is reported for `f = { }` but the code runs without exceptions

**Notes**
It seems like this corner case was handled incorrectly when fixing https://github.com/dotnet/roslyn/issues/32495
